common/v2/features/DeFiZap/components/ZapForm.tsx

```diff
@@ -1,34 +1,97 @@
 import React, { useContext } from 'react';
 import styled from 'styled-components';
 import { Formik, Form, Field, FieldProps } from 'formik';
+import { Button } from '@mycrypto/ui';
+import { isEmpty } from 'lodash';
+import * as Yup from 'yup';
 
-import { StoreContext, AssetContext, getNonce, NetworkContext, fetchGasPriceEstimates } from 'v2';
-import translate from 'v2/translations';
+import {
+  StoreContext,
+  AssetContext,
+  getNonce,
+  NetworkContext,
+  fetchGasPriceEstimates,
+  getAccountBalance
+} from 'v2';
+import translate, { translateRaw } from 'v2/translations';
+import { SPACING } from 'v2/theme';
 import { IAccount, Network, StoreAccount, Asset } from 'v2/types';
-import { Button, AccountDropdown, InlineMessage, AmountInput } from 'v2/components';
+import { AccountDropdown, InlineMessage, AmountInput } from 'v2/components';
 import { validateAmountField } from 'v2/features/SendAssets/components/validators/validators';
 import { isEthereumAccount } from 'v2/services/Store/Account/helpers';
 
 import { ZapInteractionState, ISimpleTxFormFull } from '../types';
+import ZapSelectedBanner from './ZapSelectedBanner';
+import DeFiZapLogo from './DeFiZapLogo';
+import { IZapConfig } from '../config';
+import { parseEther } from 'ethers/utils';
 
 interface Props extends ZapInteractionState {
   onComplete(fields: any): void;
   handleUserInputFormSubmit(fields: any): void;
 }
 
-const UserInputForm = styled.div`
-  margin: 0px 24px;
+interface UIProps {
+  ethAsset: Asset;
+  network: Network;
+  zapSelected: IZapConfig;
+  relevantAccounts: StoreAccount[];
+  onComplete(fields: any): void;
+}
+
+const FormFieldItem = styled.fieldset`
+  margin-bottom: ${SPACING.LG};
+`;
+
+const FormFieldLabel = styled.label`
+  display: flex;
+  font-size: 1rem;
+  margin-bottom: ${SPACING.SM};
+  font-weight: 400;
+  align-items: center;
+  flex-wrap: wrap;
+`;
+
+const FormFieldSubmitButton = styled(Button)`
+  width: 100%;
+  display: inline-block;
+  &:disabled {
+    background-color: rgba(0, 122, 153, 0.3);
+  }
+`;
+
+const DeFiZapLogoContainer = styled.div`
+  margin-top: ${SPACING.BASE};
 `;
 
 export const EtherUUID = '356a192b-7913-504c-9457-4d18c28d46e6';
```

Comment:
Isn't there a better way to do this?

Reply:
To clarify, I'm talking about the UUID line.

Reply:
I couldn't come up with anything better, you have any ideas?